x-pack/solutions/security/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_hosts.ts

```diff
@@ -377,3 +405,140 @@ export const deleteIndexedEndpointHosts = async (
 
   return response;
 };
+
+interface IndexEndpointHostForPolicyOptions {
+  esClient: Client;
+  kbnClient: KbnClient;
+  /** The Endpoint integration policy ID that should be used to index new Endpoint host */
+  integrationPolicyId: string;
+  /**
+   * The Fleet agent policy ID. By default, the first agent policy ID listed in the Integration Policy will be used,
+   * but since an Integration policy can be shared with multiple agent policies, this options can be used to target
+   * a specific one.
+   */
+  agentPolicyId?: string;
+  overrides?: DeepPartial<HostMetadataInterface>;
+  logger?: ToolingLog;
+}
+/**
+ * Indexes a new Endpoint host for a given policy (Endpoint integration policy).
+ *
+ * NOTE: consider stopping the Endpoint metadata
+ */
+export const indexEndpointHostForPolicy = async ({
+  esClient,
+  kbnClient,
+  integrationPolicyId,
+  agentPolicyId,
+  overrides = {},
+  logger = createToolingLogger(),
+}: IndexEndpointHostForPolicyOptions): Promise<IndexedHostsResponse> => {
+  const response: IndexedHostsResponse = buildIndexHostsResponse();
+  const [kibanaVersion, integrationPolicy] = await Promise.all([
+    fetchKibanaVersion(kbnClient),
+    kbnClient
+      .request<GetOnePackagePolicyResponse>({
+        path: packagePolicyRouteService.getInfoPath(integrationPolicyId),
+        method: 'GET',
+        headers: { 'elastic-api-version': '2023-10-31' },
+      })
+      .catch(catchAxiosErrorFormatAndThrow)
+      .then((res) => res.data.item),
+  ]);
+
+  logger.verbose(`Integration policy:\n${JSON.stringify(integrationPolicy, null, 2)}`);
+
+  if (agentPolicyId && !integrationPolicy.policy_ids.includes(agentPolicyId)) {
+    throw new EndpointError(
+      `indexEndpointHostForPolicy(): Invalid agent policy id [${agentPolicyId}]. Agent policy id not listed in integration policy`
+    );
+  }
+
+  const agentPolicy = await kbnClient
+    .request<GetOneAgentPolicyResponse>({
+      method: 'GET',
+      path: agentPolicyRouteService.getInfoPath(agentPolicyId ?? integrationPolicy.policy_ids[0]),
+      headers: { 'elastic-api-version': '2023-10-31' },
+    })
+    .then((res) => res.data.item);
+
+  logger.verbose(`Agent policy:\n${JSON.stringify(agentPolicy, null, 2)}`);
+
+  const timestamp = Date.now() - 3.6e6; // Subtract 1 hour
+  // hostMetadataDoc['@timestamp'] = timestamp;
+  // hostMetadataDoc.event.created = timestamp;
```

Comment:
nit: Maybe remove the commented code, unless it was needed? 🤔

Reply:
Oops. Forgot about this. Will remove it in my next commit

Reply:
Change here: https://github.com/elastic/kibana/pull/219978